### PR TITLE
Decoded wav

### PR DESCRIPTION
**Pull request by deeuu**
Following [our discussion](https://github.com/sigsep/sigsep-mus-previews/issues/6), what do you make of this:

Generate csv file for 30s previews:
```
python generate_previews.py --musb /path/to/musdb
```
Generate csv file for 10s segments from 30s previews (which are wav files):
```
python generate_previews.py --musb /path/to/previews --duration 10 --iswav
```

### CODE REVIEW

**Comment by faroit** [comment]
makes sense, but then as I used the `max` of the activity. both sets would be centered around the same start sample, meaning we could also directly output the 30s and 10s excerpts in one run. What do you think?

**Comment by faroit** [comment]
> both sets would be centered around the same start sample

I just had a look at it again, I actually did not implement a `centering`, so we should do this maybe first so that the peak is in the center of the excerpts regardless of the duration

**Comment by deeuu** [comment]
Okay, I had to look into this because I haven't quite got my head around the procedure.

It would be nice to have it how this branch suggests, making it independent of the application then (we don't have to explain why we are generating two lots of previews).

This should be fine so long as the 30s portion is centred on the peak of the full activity series (as you suggest), though I don't know the details of the framing implementation.

Basically, we want:
10s segment from entire audio == 10s segment from 30s segment from entire audio.

**Comment by faroit** [comment]
>  10s segment from entire audio == 10s segment from 30s segment from entire audio.

exactly. But we have one more constraint which is that all excerpts need to be quantised to integer seconds so match with the bsseval.

for example:
```
30s: ______^______
10s:      _^__
```

the 10s segment could be slightly off centered regarding the activity peak compared to the 30s.
So we need:

* [ ] centering around the peak position
* [ ] get sample level start and end based on center pos
* [X] quantize both start end to second level

**Comment by deeuu** [comment]
Does [longterm_win](https://github.com/sigsep/sigsep-mus-previews/blob/master/generate_previews.py#L154) need to depend on the preview length? If it were fixed, this might not be an issue?

**Comment by deeuu** [comment]
Looking more at this, I think you are right: it is not easy to align the centres without running the extractions at the same time (based on the same centre sample).

 I propose, however, to keep these independent, but make sure that the centring is correct.

That sound ok?

**Comment by faroit** [comment]
> Does longterm_win need to depend on the preview length? If it were fixed, this might not be an issue?

can be fixed

> Looking more at this, I think you are right: it is not easy to align the centres without running the extractions at the same time (based on the same centre sample).

> I propose, however, to keep these independent, but make sure that the centring is correct.

> That sound ok?

👍

**Comment by deeuu** [comment]
@faroit Have gone through the code in more detail, and I don't see an issue now. The start and end points are [determined by the desired preview length](https://github.com/sigsep/sigsep-mus-previews/blob/master/generate_previews.py#L156), which makes sense to me.

Thus, the definition of peak activity is dependent on the preview size (long-term activity averaging), so we can't do:
> 10s segment from entire audio == 10s segment from 30s segment from entire audio.

I'm happy to just extract the 10s segments from the 30s previews used for the website, even if they aren't the 'optimal' selection.

Since the quantization is already implemented, both sets of previews should be aligned with the BSS Eval measures (right?).
